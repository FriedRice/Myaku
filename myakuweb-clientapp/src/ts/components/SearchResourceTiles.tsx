/**
 * @module Additional resource tiles to accompy with a search result component
 */

import React from 'react';
import ResourceLinkSetTile from './ResourceLinkSetTile';
import { SearchResources } from '../types';
import Tile from './Tile';

interface SearchResourceTilesProps {
    resources: SearchResources;
}
type Props = SearchResourceTilesProps;


const SearchResourceTiles: React.FC<Props> = function(props) {
    var linkSetTiles: React.ReactElement[] = [];
    for (const linkSet of props.resources.resourceLinkSets) {
        linkSetTiles.push(
            <ResourceLinkSetTile
                key={linkSet.setName}
                query={props.resources.query}
                linkSet={linkSet}
            />
        );
    }

    return (
        <aside className='resource-links-aside'>
            <Tile tileClasses='aside-tile resource-header-tile'>
                <h4>More Resources</h4>
            </Tile>
            {linkSetTiles}
        </aside>
    );
};

export default SearchResourceTiles;
