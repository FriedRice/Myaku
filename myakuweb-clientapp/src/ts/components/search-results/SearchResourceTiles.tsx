/**
 * @module Additional resource tiles to accompy with a search result component
 */

import React from 'react';
import ResourceLinkSetTile from
    'ts/components/search-results/ResourceLinkSetTile';
import { SearchResources } from 'ts/types/types';
import Tile from 'ts/components/generic/Tile';

interface SearchResourceTilesProps {
    resources: SearchResources;
}
type Props = SearchResourceTilesProps;


const SearchResourceTiles: React.FC<Props> = function(props) {
    var linkSetTiles: React.ReactElement[] = [];
    for (const linkSet of props.resources.resourceLinkSets) {
        linkSetTiles.push(
            <ResourceLinkSetTile
                key={linkSet.setName}
                query={props.resources.query}
                linkSet={linkSet}
            />
        );
    }

    return (
        <aside className='resource-links-aside'>
            <Tile tileClasses='aside-tile resource-header-tile'>
                <h4>More Resources</h4>
            </Tile>
            {linkSetTiles}
        </aside>
    );
};

export default SearchResourceTiles;
