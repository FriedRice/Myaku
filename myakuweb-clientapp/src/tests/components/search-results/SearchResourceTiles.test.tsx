/**
 * Tests for the [[SearchResourceTiles]] component.
 */

import React from 'react';
import SearchResourceTiles from
    'ts/components/search-results/SearchResourceTiles';
import { shallow } from 'enzyme';

import {
    SEARCH_RESOURCES_NO,
    SEARCH_RESOURCES_OB,
} from 'tests/testData';


describe('<SearchResourceTiles />', function() {
    it('renders correctly with resources set', function() {
        const wrapper = shallow(
            <SearchResourceTiles
                resources={SEARCH_RESOURCES_NO}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({resources: SEARCH_RESOURCES_OB});
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with no resources set', function() {
        const wrapper = shallow(
            <SearchResourceTiles
                resources={null}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({resources: SEARCH_RESOURCES_NO});
        expect(wrapper).toMatchSnapshot();
    });
});
