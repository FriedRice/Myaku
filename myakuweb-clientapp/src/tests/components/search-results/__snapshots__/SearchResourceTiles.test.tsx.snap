// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResourceTiles /> renders correctly with no resources set 1`] = `
<aside
  className="resource-links-aside"
>
  <Tile
    tileClasses="aside-tile resource-header-tile"
  >
    <h4>
      More Resources
    </h4>
  </Tile>
  <ResourceLinkSetTile
    key="0"
    linkSet={null}
    query={null}
  />
  <ResourceLinkSetTile
    key="1"
    linkSet={null}
    query={null}
  />
  <ResourceLinkSetTile
    key="2"
    linkSet={null}
    query={null}
  />
</aside>
`;

exports[`<SearchResourceTiles /> renders correctly with no resources set 2`] = `
<aside
  className="resource-links-aside"
>
  <Tile
    tileClasses="aside-tile resource-header-tile"
  >
    <h4>
      More Resources
    </h4>
  </Tile>
  <ResourceLinkSetTile
    key="Jpn-Eng Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://jisho.org/search/%E3%81%AE",
            "resourceName": "Jisho.org",
          },
          Object {
            "link": "https://ejje.weblio.jp/content/%E3%81%AE",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Dictionaries",
      }
    }
    query="の"
  />
  <ResourceLinkSetTile
    key="Jpn-Eng Sample Sentences"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://tatoeba.org/eng/sentences/search?query=%3D%E3%81%AE&from=jpn&to=eng",
            "resourceName": "Tatoeba",
          },
          Object {
            "link": "https://ejje.weblio.jp/sentence/content/%22%E3%81%AE%22",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Sample Sentences",
      }
    }
    query="の"
  />
  <ResourceLinkSetTile
    key="Jpn Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://dictionary.goo.ne.jp/srch/all/%E3%81%AE/m1u/",
            "resourceName": "Goo",
          },
          Object {
            "link": "https://www.weblio.jp/content/%E3%81%AE",
            "resourceName": "Weblio",
          },
        ],
        "setName": "Jpn Dictionaries",
      }
    }
    query="の"
  />
</aside>
`;

exports[`<SearchResourceTiles /> renders correctly with resources set 1`] = `
<aside
  className="resource-links-aside"
>
  <Tile
    tileClasses="aside-tile resource-header-tile"
  >
    <h4>
      More Resources
    </h4>
  </Tile>
  <ResourceLinkSetTile
    key="Jpn-Eng Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://jisho.org/search/%E3%81%AE",
            "resourceName": "Jisho.org",
          },
          Object {
            "link": "https://ejje.weblio.jp/content/%E3%81%AE",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Dictionaries",
      }
    }
    query="の"
  />
  <ResourceLinkSetTile
    key="Jpn-Eng Sample Sentences"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://tatoeba.org/eng/sentences/search?query=%3D%E3%81%AE&from=jpn&to=eng",
            "resourceName": "Tatoeba",
          },
          Object {
            "link": "https://ejje.weblio.jp/sentence/content/%22%E3%81%AE%22",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Sample Sentences",
      }
    }
    query="の"
  />
  <ResourceLinkSetTile
    key="Jpn Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://dictionary.goo.ne.jp/srch/all/%E3%81%AE/m1u/",
            "resourceName": "Goo",
          },
          Object {
            "link": "https://www.weblio.jp/content/%E3%81%AE",
            "resourceName": "Weblio",
          },
        ],
        "setName": "Jpn Dictionaries",
      }
    }
    query="の"
  />
</aside>
`;

exports[`<SearchResourceTiles /> renders correctly with resources set 2`] = `
<aside
  className="resource-links-aside"
>
  <Tile
    tileClasses="aside-tile resource-header-tile"
  >
    <h4>
      More Resources
    </h4>
  </Tile>
  <ResourceLinkSetTile
    key="Jpn-Eng Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://jisho.org/search/OB",
            "resourceName": "Jisho.org",
          },
          Object {
            "link": "https://ejje.weblio.jp/content/OB",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Dictionaries",
      }
    }
    query="OB"
  />
  <ResourceLinkSetTile
    key="Jpn-Eng Sample Sentences"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://tatoeba.org/eng/sentences/search?query=%3DOB&from=jpn&to=eng",
            "resourceName": "Tatoeba",
          },
          Object {
            "link": "https://ejje.weblio.jp/sentence/content/%22OB%22",
            "resourceName": "Weblio EJJE",
          },
        ],
        "setName": "Jpn-Eng Sample Sentences",
      }
    }
    query="OB"
  />
  <ResourceLinkSetTile
    key="Jpn Dictionaries"
    linkSet={
      Object {
        "resourceLinks": Array [
          Object {
            "link": "https://dictionary.goo.ne.jp/srch/all/OB/m1u/",
            "resourceName": "Goo",
          },
          Object {
            "link": "https://www.weblio.jp/content/OB",
            "resourceName": "Weblio",
          },
        ],
        "setName": "Jpn Dictionaries",
      }
    }
    query="OB"
  />
</aside>
`;
